app/controllers/api/deploys_controller.rb

```diff
@@ -1,16 +1,43 @@
 # frozen_string_literal: true
 class Api::DeploysController < Api::BaseController
-  include CurrentProject
-
   skip_before_action :require_project, only: [:active_count]
 
   def index
-    scope = current_project.try(:deploys) || Deploy
-    @deploys = paginate(params[:ids].present? ? [scope.find(params[:ids])].flatten : scope)
-    render json: @deploys
+    render json: paginate(deploy_scope)
   end
 
   def active_count
     render json: Deploy.active.count
   end
+
+  protected
+
+  def filter
+    if filter_param = params.fetch(:filter, false)
+      filter_param if ["pending", "running", "canceled", "failed", "errored", "succeeded"].include?(filter_param)
```

Comment:
also should be bad_request if it is not included instead of returning all silently